common/flatpak-dir.c

```diff
@@ -4492,9 +4496,13 @@ flatpak_dir_create_system_child_repo (FlatpakDir   *self,
   repo_dir_config = g_file_get_child (repo_dir, "config");
   if (!g_file_query_exists (repo_dir_config, NULL))
     {
-      if (!ostree_repo_create (new_repo,
-                               OSTREE_REPO_MODE_BARE_USER,
-                               NULL, error))
```

Comment:
Hmmm, actually I wonder if this could be a generic problem even if the system repo is bare-user-only. It will work for "correctly" built flatpaks (some changes i added, recently) but not for all historical ones.

I think the right approach for now is to make this code copy the system repo type, and then make the other commit optionally use bare-user-only. Then we can eventually switch it to that by default.